### PR TITLE
[Editor] Add close button to conditional breakpoint panel

### PR DESCRIPTION
**Pull request by ryanjduffy**
Associated Issue: #1945

### Summary of Changes

* Using the 'x' saves and closes the panel
* Pressing "Escape" closes without saving
* Pressing "Enter" always saves. If there is no content, it converts to a "regular" breakpoint

#### Save on Close Button

- [x] Open a source file
- [x] Right click gutter to "Add conditional breakpoint"
- [x] Enter a breakpoint (e.g. `console.log('test')`)
- [x] Clicking "x" closes the panel
- [x] Right click breakpoint to "Edit breakpoint"
- [x] Conditional breakpoint panel should contain specified condition

#### Save on Enter

- [x] Open a source file
- [x] Right click gutter to "Add conditional breakpoint"
- [x] Enter a breakpoint (e.g. `console.log('test')`)
- [x] Press <Enter> key
- [x] Right click breakpoint to "Edit breakpoint"
- [x] Conditional breakpoint panel should contain specified condition

#### Cancel add on Escape

- [x] Open a source file
- [x] Right click gutter to "Add conditional breakpoint"
- [x] Enter a breakpoint (e.g. `console.log('test')`)
- [x] Press <Escape> key
- [x] No breakpoint should be created

#### Cancel edit on Escape

- [x] Open a source file
- [x] Right click gutter to "Add conditional breakpoint"
- [x] Enter a breakpoint (e.g. `console.log('test')`)
- [x] Press <Enter> key or click close button
- [x] Right click breakpoint to "Edit breakpoint"
- [x] Edit breakpoint (e.g. `console.log('updated')`)
- [x] Press <Escape> key
- [x] Right click breakpoint to "Edit breakpoint"
- [x] Breakpoint should contain original condition

#### Convert to normal breakpoint

- [x] Open a source file
- [x] Right click gutter to "Add conditional breakpoint"
- [x] Enter a breakpoint (e.g. `console.log('test')`)
- [x] Press <Enter> key or click close button
- [x] Right click breakpoint to "Edit breakpoint"
- [x] Remove condition
- [x] Press <Enter> key or click close button
- [x] Breakpoint should render as normal breakpoint

### Screenshots/Videos

closebutton

### CODE REVIEW

File: src/components/Editor/ConditionalPanel.js [hunk 1]
```diff
@@ -28,7 +39,12 @@ function renderConditionalPanel({ condition, closePanel, setBreakpoint }:
       dom.input({
         defaultValue: condition,
         placeholder: L10N.getStr("editor.conditionalPanel.placeholder"),
-        onKeyPress: onKey
+        onKeyDown: onKey,
+        ref: setInput
+      }),
+      CloseButton({
+        handleClick: saveAndClose,
+        buttonClass: "big"
```
Comment (irfanhudda) [inline src/components/Editor/ConditionalPanel.js]: Might be good to add tooltip here using `tooltip` prop to this component.

Comment (ryanjduffy) [inline src/components/Editor/ConditionalPanel.js]: Fair point - updated!

**Comment by ryanjduffy** [comment]
👍

**Comment by jasonLaster** [comment]
> using the 'x' saves and closes the panel

i would guess that `x`  would cancel and close the panel.

**Comment by irfanhudda** [comment]
@jasonLaster Tooltip should explain the behaviour here. I'm fine with `x` being save and close or cancel and close.

**Comment by jasonLaster** [comment]
Hmm, lets do "cancel" if there are no strong objections:

we can revisit in a couple weeks… and the cost of canceling is less significant than saving something bad.

**Comment by ryanjduffy** [comment]
👍

**Comment by irfanhudda** [comment]
@ryanjduffy Thanks for this PR.

**Pull request merged**